service/src/main/java/zone/cogni/asquare/service/index/IndexService.java

```diff
@@ -345,4 +345,82 @@ private void reindexIteration(List<ResourceIndex> resources, List<ResourceIndex>
     log.info("Indexing iteration is done. Duration: {}, failed resources: {}", IndexUtils.prettyDurationMs(System.currentTimeMillis() - startIteration), failedResources.size() + interruptedGraphs.size());
   }
 
+  /**
+   * This method is an alternative to reindexSync/reindexAsync
+   * but leaves the control of the sync/async selection,
+   * as well as the executor, to the code that uses it.
+   *
+   * @param rdfStoreService the service used for the desired triple store
+   * @param sparqlQuery     the SPARQL query that selects the resources to be indexed
+   * @return Map<String, Long> with information about the indexing results
+   */
+  public Map<String, Long> reindexSimple(RdfStoreService rdfStoreService, String sparqlQuery) {
+    log.info("Launching reindexing for query {}", sparqlQuery);
+    List<ResourceIndex> resources = Collections.synchronizedList(findAllIndexResources(sparqlQuery, rdfStoreService));
+    log.info("Query finished with {} results", resources.size());
+    return reindexSimpleExecution(resources, rdfStoreService);
+  }
+
+  private Map<String, Long> reindexSimpleExecution(List<ResourceIndex> resources, RdfStoreService rdfStoreService) {
```

Comment:
this method seems duplicating existing reindexSync with only differene that it is executing  reindexSimpleIteration instead of reindexIteration . This could be parametrized isnt it?

Smt like `isSimple ? reindexSimpleIteration(resources, failedResources, rdfStoreService, timestamp) : reindexIteration(resources, failedResources, rdfStoreService, timestamp);`

Reply:
I am not a huge fan of the boolean tbh. nor of the 2 similar flows to be fair though..